docs/developer/release-process.md

```diff
@@ -97,49 +97,47 @@ helm dependency update ./chart/kubeapps
 
 Currently, we have three types of dependencies: the [dashboard dependencies](../../dashboard/package.json), the [golang dependencies](../../go.mod), and the [rust dependencies](../../cmd/pinniped-proxy/Cargo.toml). They _must_ be upgraded to the latest minor/patch version to get the latest bug and security fixes.
 
-- Upgrade the [dashboard dependencies](../../dashboard/package.json) by running:
+#### Dashboard dependencies
+
+Upgrade the [dashboard dependencies](../../dashboard/package.json) by running:
 
 ```bash
 cd dashboard
 yarn upgrade
 ```
 
-- Check the outdated [golang dependencies](../../go.mod) as explained in the Golang documentation on [how to upgrade and downgrade dependencies](https://github.com/golang/go/wiki/Modules#how-to-upgrade-and-downgrade-dependencies).
-  First, clean and tidy up any unused dependencies and list the upgradeable ones.
-  Then, try to manually update those versions that can be safely upgraded. A useful tool for doing so is [go-mod-upgrade](https://github.com/oligot/go-mod-upgrade).
-  Alternatively, you can run the `go get -u ...` command to perform an automatic upgrade.
+#### Golang dependencies
+
+Check the outdated [golang dependencies](../../go.mod) by running the following (from [How to upgrade and downgrade dependencies](https://github.com/golang/go/wiki/Modules#how-to-upgrade-and-downgrade-dependencies)):
 
 ```bash
 go mod tidy
 go list -u -f '{{if (and (not (or .Main .Indirect)) .Update)}}{{.Path}}: {{.Version}} -> {{.Update.Version}}{{end}}' -m all 2> /dev/null
-go-mod-upgrade
-# go get -u -t ./... # upgrades to minor or patch releases
-# go get -u=patch  -t ./... # upgrades just to patch releases
+go get -u ./...
```

Comment:
Deja vu, but doesn't the above already try to "upgrade to the latest version for all direct and indirect dependencies of the current module", in which case, the following sentence doesn't make sense. I think we should either be recommending doing `go get -u ./...` and dealing with the fallout manually (won't this try to upgrade dependencies that we can't yet upgrade? Fails for me locally with `module declares its path as: github.com/moby/spdystream but was required as: github.com/docker/spdystream`, but that could be transitional), or recommending doing `go-mod-upgrade` and selecting the packages we want to upgrade and leaving those that we can't?

Reply:
Yes, you're right `go get -u ./...` would perform the upgrade (usually), however, there are some cases (as you faced) in which it is not possible. In that case, the `go-mod-upgrade` could help, as it just updates the ones you select.
I'll try to reword it; I agree currently is quite confusing, sorry :S